lxd/device/nic_routed.go

```diff
@@ -110,6 +112,33 @@ func (d *nicRouted) validateConfig(instConf instance.ConfigReader) error {
 		}
 	}
 
+	for _, keyPrefix := range []string{"ipv4", "ipv6"} {
+		addr, ok := d.config[fmt.Sprintf("%s.address", keyPrefix)]
+		if ok {
+			p, err := ping.NewPinger(addr)
```

Comment:
I believe there was an ability to ping out of a specific interface, if so then we should use that and set it to the parent interface. Also if there is no parent set, then we should not do the ping detection at all, as we won't be advertising via arp/ndp the address anyway.

Reply:
I can't see a way to ping out of a specific interface with this library (I changed the library so that we could use the same one for IPv4 and v6). I can revert to the arping library and find another library for IPv6 if that would be best?

Reply:
OK I didn't realise that, in that case no problem

Reply:
I've now implemented this. Checks only occur if the parent interface is set and the arp/ndp uses the specific interface for each check.